src/ui/Forms/SpellCheck.cs

```diff
@@ -659,40 +646,39 @@ public bool DoSpell(string word)
             return _hunspell.Spell(word);
         }
 
-        public List<string> DoSuggest(string word)
+        /// <summary>
+        /// Asynchronously suggests alternative words for a misspelled word using Hunspell spell-checker.
+        /// </summary>
+        /// <param name="word">The misspelled word to suggest alternatives for.</param>
+        /// <returns>A task that represents the asynchronous operation. The task result contains a list of suggested alternative words for the misspelled word.</returns>
+        private async Task<List<string>> DoSuggestAsync(string word)
         {
-            var parameter = new SuggestionParameter(word, _hunspell);
-            var suggestThread = new System.Threading.Thread(DoWork);
-            suggestThread.Start(parameter);
-            suggestThread.Join(3000); // wait max 3 seconds
-            if (!parameter.Success)
+            try
+            {
+                return await Task.Run(() => _hunspell.Suggest(word), CancellationToken.None).ConfigureAwait(false);
```

Comment:
It's safe to resume on a thread-pool thread here since no interaction with the UI was done from this point!